src/pymovements/measure/measures.py

```diff
@@ -50,11 +52,111 @@ def null_ratio(column: str, column_dtype: pl.DataType) -> pl.Expr:
         value = 1 - pl.col(column).count() / pl.col(column).len()
     elif column_dtype == pl.List:
         non_null_lengths = pl.col(column).list.drop_nulls().drop_nans().list.len()
-        value = 1 - (non_null_lengths == pl.col(column).list.len()).sum() / pl.col(column).len()
+        value = (
+            1 - (non_null_lengths == pl.col(column).list.len()).sum() /
+            pl.col(column).len()
+        )
     else:
         raise TypeError(
             'column_dtype must be of type {Float64, Int64, Utf8, List}'
             f' but is of type {column_dtype}',
         )
 
     return value.alias('null_ratio')
+
+
+@register_sample_measure
+def data_loss(
+        timestamp_col: str = 'time',
+        *,
+        start_time: float | None = None,
+        end_time: float | None = None,
+        unit: Literal['count', 'time', 'ratio'] = 'count',
+) -> pl.Expr:
+    """Measure gaze data loss within a group using only Polars expressions.
+
+    Computes the number of missing samples ("count"), missing time ("time"),
+    or the ratio of missing to expected samples ("ratio"). The expected
+    sampling is derived from the group's median inter-sample interval (ISI)
```

Comment:
I'm afraid that this introduces circular logic:

when you automatically derive the sampling rate from the samples, and then use this as a basis to calculate the expected samples, then, if you disregard null/nan values, it will always result in the top value (100%).

It is better to explicitly add `sampling_rate: float` as an argument to `data_loss()`. The way data loss is usually defined the measure can only be applied to evenly sampled time series (or rather to a time series that is expected to be evenly sampled).

A related measure to your inference logic is the *effective sampling rate* (N / delta time) which is also very useful but probably out of scope of this PR.